internal/tiltfile/tiltfile_state.go

```diff
@@ -204,6 +206,14 @@ to your Tiltfile. Otherwise, switch k8s contexts and restart Tilt.`, kubeContext
 	}
 	manifests = append(manifests, localManifests...)
 
+	flagsState, _ := flags.GetState(result)
+	if len(flagsState.Resources) > 0 {
+		requested = nil
+		for _, r := range flagsState.Resources {
```

Comment:
It might make more sense for this machinery to be a method of flags state? Or better yet, make flag state store manifestnames so you don't need to do this conversion at all?

Reply:
yeah, that makes this cleaner